snapstack/base.py

```diff
@@ -0,0 +1,93 @@
+from collections import OrderedDict
+
+from snapstack.config import CONF_FILES
+from snapstack.step import Step
+
+
+class Base:
+    def __init__(self):
+        '''
+        Setup a private OrderedDict to store steps.
+
+        '''
+        self._steps = OrderedDict()
+
+    def steps(self):
+        '''
+        Return a list of Step objects, in order.
+
+        '''
+        return [s for _, s in self._steps.items()]
+
+    def add_steps(self, *steps):
+        '''
+        Each step arg should be a tuple, comprising a name as its first
+        element, and a Step object as its second.
+
+        '''
+        for name, value in steps:
+            self._steps[name] = value
+
+    def remove_steps(self, *steps):
+        '''
+        Each arg should be a stirng naming a step. We'll remove the named
+        step from our map of steps.
+
+        '''
+        for name in steps:
+            del self._steps[name]
+
+
+class Setup(Base):
+    def __init__(self):
+        super(Setup, self).__init__()
+        self._steps['confs'] = Step(
+            script_loc='{snap-test}',
+            files=CONF_FILES
+        )
+        self._steps['packages'] = Step(
+            script_loc='{snapstack}',
+            tests=['packages.sh']
```

Comment:
I'd be somewhat tempted to rename tests to scripts, making it a bit more generic because it can contain setup/cleanup scripts as well as test scripts.